lib/v1alpha/network.go

```diff
@@ -117,38 +127,89 @@ func GetInterfaces() (*types.InterfaceResponse, error) {
-		if interfaceData.Type == types.INTERFACE_TYPE_UNKNOWN {
-			if len(interfaceData.Network.Gateway) > 0 {
-				glog.Infof(" Set Interface to kernel: %s", interfaceData.IfName)
-				interfaceData.Type = types.INTERFACE_TYPE_KERNEL
-			} else {
-				// TBD: Need a better way to determine if SR_IOV or not.
-				glog.Infof(" Set Interface to SR-IOV: %s  PCIIndex=%d Len=%d", interfaceData.IfName, pciIndex, len(pciAddressSlice))
-				interfaceData.Type = types.INTERFACE_TYPE_SRIOV
-				if pciIndex < len(pciAddressSlice) {
-					interfaceData.Sriov = &types.SriovData{
-						PciAddress: pciAddressSlice[pciIndex],
+	for _, ifaceData := range response.Interface {
+		if ifaceData.DeviceType == types.INTERFACE_TYPE_UNKNOWN {
+			if ifaceData.NetworkStatus.Default {
+				ifaceData.DeviceType = types.INTERFACE_TYPE_HOST
+				glog.Infof("%s is the \"default\" interface, mark as \"%s\"",
+					ifaceData.NetworkStatus.Interface, ifaceData.DeviceType)
+			} else if pciIndex < len(pciAddressSlice) {
+				if ifaceData.NetworkStatus.DeviceInfo == nil {
+					ifaceData.DeviceType = types.INTERFACE_TYPE_SRIOV
+					ifaceData.NetworkStatus.DeviceInfo = &nettypes.DeviceInfo{
+						Type:    nettypes.DeviceInfoTypePCI,
+						Version: nettypes.DeviceInfoVersion,
+						Pci: &nettypes.PciDevice{
+							PciAddress: pciAddressSlice[pciIndex],
+						},
 					}
 					pciIndex++
-					glog.Infof(" Added PCI Address: %s", interfaceData.Sriov.PciAddress)
+					glog.Infof("%s was \"unknown\", mark as \"%s\"",
+						ifaceData.NetworkStatus.Interface, ifaceData.DeviceType)
 				} else {
-					glog.Warningf("More SR-IOV interfaces detected than PCI addresses - %s", interfaceData.IfName)
+					glog.Warningf("%s was \"unknown\", but DeviceInfo exists with type \"%s\"",
+						ifaceData.NetworkStatus.Interface, ifaceData.NetworkStatus.DeviceInfo.Type)
 				}
+			} else {
+				ifaceData.DeviceType = types.INTERFACE_TYPE_HOST
+				glog.Infof("%s was \"unknown\", mark as \"%s\"",
+					ifaceData.NetworkStatus.Interface, ifaceData.DeviceType)
+			}
+		}
+	}
+
+	// PCI Address found that did not match an existing interface in the
+	// NetworkStatus annotation so add to list.
+	if pciIndex < len(pciAddressSlice) {
+		for _, pciAddr := range pciAddressSlice[pciIndex:] {
+			ifaceData := &types.InterfaceData{
+				DeviceType: types.INTERFACE_TYPE_SRIOV,
+				NetworkStatus: nettypes.NetworkStatus{
+					DeviceInfo: &nettypes.DeviceInfo{
+						Type:    nettypes.DeviceInfoTypePCI,
+						Version: nettypes.DeviceInfoVersion,
+						Pci: &nettypes.PciDevice{
+							PciAddress: pciAddr,
+						},
+					},
+				},
 			}
+			response.Interface = append(response.Interface, ifaceData)
+
+			glog.Infof("Adding %s as new interface because no other matches, type \"%s\"",
+				pciAddr, ifaceData.DeviceType)
 		}
 	}
 
 	glog.Infof("RESPONSE:")
-	for _, interfaceData := range response.Interface {
-		glog.Infof("%v", interfaceData)
+	for _, ifaceData := range response.Interface {
```

Comment:
I think there might still be devices that has `empty` type at this point, right? it would be the device that has a deviceInfo but DeviceInfo.Type is not configured.

Reply:
Yes, what should I do with those? Invalid config.

Reply:
nothing need to change, I just make sure understand it correctly.